Formula/mesos.rb

```diff
@@ -53,6 +53,16 @@ class Mesos < Formula
   def install
     ENV.java_cache
 
+    # Disable optimizing as libc++ does not play well with optimized clang
+    # builds (see https://llvm.org/bugs/show_bug.cgi?id=28469 and
+    # https://issues.apache.org/jira/browse/MESOS-5745).
+    #
+    # NOTE: We cannot use `--disable-optimize` since we also pass e.g.,
+    # CXXFLAGS via environment variables. Since compiler flags are passed via
+    # environment variables the Mesos build will silently ignore flags like
+    # `--[disable|enable]-optimize`.
+    ENV.O0
```

Comment:
@ilovezfs: I am not able to build this with either gcc-5 or gcc-6 (fails in bundled 3rdparty tools) so I believe making this conditional on clang is not that useful in the long run. Also, the real issue is with the used stdlib, not with the compiler, and theoretically users might build using clang with libstdc++, or with gcc and libcxx (not sure if that's a practical concern for Homebrew).

@MikeMcQuaid: Re:guarding on the clang version: AFAIK Apple assigns its clang version numbers kind of arbitrary, and there is e.g., no guarantee that clang-3.9 would be the basis of Apple clang-9.0.0 (I think it is not yet released).

I guess we should try to fix https://issues.apache.org/jira/browse/MESOS-6606 upstream (e.g., to gracefully degrade on setups with this broken libcxx); after that we should be able to remove the non-standard configuration here.